components/server/src/redis/data-cache.ts

```diff
@@ -0,0 +1,53 @@
+/**
+ * Copyright (c) 2023 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License.AGPL.txt in the project root for license information.
+ */
+
+import { keyPattern } from "@gitpod/gitpod-db/lib/data-cache";
+import { inject, injectable } from "inversify";
+import { RedisClient } from "./client";
+import { cacheHitsCounter, cacheLatencyHistogram, cacheMissesCounter } from "../prometheus-metrics";
+import { log } from "@gitpod/gitpod-protocol/lib/util/logging";
+
+const TTL_SEC = 5 * 60; // 5 minutes
+
+@injectable()
+export class DataCacheRedis implements DataCache {
+    @inject(RedisClient) protected redis: RedisClient;
+
+    async get<T>(key: string[], provider: () => Promise<T | undefined>): Promise<T | undefined> {
+        const cacheKey = key.join(":");
+        const stop = cacheLatencyHistogram.startTimer();
+        let result: string | null = null;
+        try {
+            result = await this.redis.get().get(cacheKey);
+        } catch (error) {
+            log.error("Error retrieving cache value", error, { cacheKey });
+        }
+        stop();
+        if (result) {
+            cacheHitsCounter.inc();
+            return JSON.parse(result);
+        } else {
+            cacheMissesCounter.inc();
+            const value = await provider();
+            if (value) {
+                try {
+                    await this.redis.get().setex(cacheKey, TTL_SEC, JSON.stringify(value));
```

Comment:
`As of Redis version 2.6.12, this command is regarded as deprecated.`

Reply:
https://redis.io/commands/setex/

Reply:
It is not deprecated on the client library, nor does the `set` function allow the `ex` argument.

Reply:
https://luin.github.io/ioredis/classes/Redis.html#set

<img width="808" alt="Screenshot 2023-06-13 at 13 21 33" src="https://github.com/gitpod-io/gitpod/assets/1419286/a45bc7b2-4a6b-4d59-ae09-7d8a02cbd17a">